starlark/testdata/set.star

```diff
@@ -116,3 +116,27 @@ assert.eq(iter(), [1, 2, 3])
 
 # sets are not indexable
 assert.fails(lambda : x[0], "unhandled.*operation")
+
+# adding and removing
+x.add(4)
+assert.true(4 in x)
+
+# remove 
+x.remove(4)
+assert.true(4 not in x)
+assert.fails(lambda: x.remove(4), "remove: missing key")
+
+# discard
+x.add(4)
+x.discard(4)
+assert.true(4 not in x)
+assert.eq(x.discard(4), None)
+
+# pop
+z = set([1])
+assert.eq(z.pop(), 1)
+assert.fails(lambda: z.pop(), "pop: empty set")
+
+# clear
+x.clear()
```

Comment:
Can you add tests of each operation on a frozen set (see dict.star for the template), including the edge cases where a potentially mutating operation does not attempt to mutate the set?